lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -183,19 +183,26 @@ public function listTableIndexes($table)
     /**
      * Returns true if all the given tables exist.
      *
-     * @param string[] $tableNames
+     * To avoid BC-break not all parameters are included.
+     *
+     * @param string|string[] $tableNames
+     * @param bool            $filtered   Optional parameter, if false ignores filters.
      *
      * @return bool
      */
     public function tablesExist($tableNames)
     {
         $tableNames = array_map('strtolower', (array) $tableNames);
 
-        return count($tableNames) === count(array_intersect($tableNames, array_map('strtolower', $this->listTableNames())));
+        $args = func_get_args();
```

Comment:
As mentioned above, using separate methods for checking existence in current DB or all DBs would be best